### PR TITLE
Test if dapr path exists, throw exception if it doesnt

### PR DESCRIPTION
**Pull request by onionhammer**
Addresses #1211 

Also adds method where more possible dapr paths can be added

### CODE REVIEW

File: src/Aspire.Hosting.Dapr/DaprDistributedApplicationLifecycleHook.cs [hunk 1]
```diff
@@ -45,6 +40,10 @@ public async Task BeforeStartAsync(DistributedApplicationModel appModel, Cancell
 
         var sideCars = new List<ExecutableResource>();
 
+        var fileName = this._options.DaprPath
+            ?? GetDefaultDaprPath()
+            ?? throw new DistributedApplicationException("Unable to locate Dapr executable.");
```
Comment (philliphoff) [inline src/Aspire.Hosting.Dapr/DaprDistributedApplicationLifecycleHook.cs:45]: Nit: I might use "Unable to locate the Dapr CLI." which could help differentiate between it and the Dapr runtime (i.e. `daprd`).  The two are sometimes confused.

**Review by philliphoff — APPROVED** [review]

**Comment by onionhammer** [comment]
@philliphoff what's the status on this? Do you need anything from me?

**Review by davidfowl — APPROVED** [review]

**Review by philliphoff — APPROVED** [review]
Looks good!

**Comment by danmoseley** [comment]
@philliphoff can we merge this ? I'll rerun validation

**Comment by philliphoff** [comment]
@danmoseley Yes, assuming validation passes, please merge.

**Comment by danmoseley** [comment]
thanks @onionhammer !

**Comment by onionhammer** [comment]
Boo, scrubbed from the release notes :P

**Comment by davidfowl** [comment]
We can fix that 😄 

@joperezr ?

**Comment by joperezr** [comment]
So sorry about that @onionhammer, we used the automatic `Generate Notes` feature from GitHub, and while it did list out your PR it didn't listed you for some reason in the New contributors list, I've manually fixed that now :)

**Pull request merged**